### PR TITLE
add snapshot support for structure selections

### PR DESCRIPTION
**Pull request by arose**
# Description

Add snapshot support for structure selections. Off by default.

## Actions

- [x] Added description of changes to the `[Unreleased]` section of `CHANGELOG.md`
- [x] Updated headers of modified files

### CODE REVIEW

File: src/mol-plugin-state/manager/structure/selection.ts [hunk 1]
```diff
@@ -484,6 +491,31 @@ export class StructureSelectionManager extends StatefulPluginComponent<Structure
         }
     }
 
+    getSnapshot(): StructureSelectionSnapshot {
+        const entries: StructureSelectionSnapshot['entries'] = [];
+
+        this.entries.forEach((entry, ref) => {
+            entries.push({
+                ref,
+                bundle: StructureElement.Bundle.fromLoci(entry.selection)
```
Comment (dsehnal) [inline src/mol-plugin-state/manager/structure/selection.ts:500]: Could it happen that the bundle is using a typed array? In that case should convert it to native array.

File: src/mol-plugin/state.ts [hunk 2]
```diff
@@ -89,6 +92,9 @@ class PluginState extends PluginComponent {
         if (snapshot.structureFocus) {
             this.plugin.managers.structure.focus.setSnapshot(snapshot.structureFocus);
         }
+        if (snapshot.structureSelection) {
+            this.plugin.managers.structure.selection.setSnapshot(snapshot.structureSelection);
```
Comment (dsehnal) [inline src/mol-plugin/state.ts:96]: Will this work on fresh snapshots where the visual isn't created yet?

File: src/mol-plugin-state/manager/structure/selection.ts (hunk 1 above)
Comment (arose) [reply src/mol-plugin-state/manager/structure/selection.ts:500]: We don't prevent it with type checking but the `Bundle.from*` methods don't use typed arrays. Got an idea to enforce that easily? We already include `Bundle`s in the state in `StructureSelectionFromBundle`, `OverpaintStructureRepresentation3DFromBundle` and other transforms.

File: src/mol-plugin/state.ts (hunk 2 above)
Comment (arose) [reply src/mol-plugin/state.ts:96]: It needs the structures the bundles are refering to in the state tree which is the case here with the data snapshot applied before. Visuals can be created later and get the selection mark applied like normally. Also it worked for me using it :)

File: src/mol-plugin-state/manager/structure/selection.ts (hunk 1 above)
Comment (dsehnal) [reply src/mol-plugin-state/manager/structure/selection.ts:500]: Only to traverse the tree. But yeah, been using it in the state tree already, so should be fine.

**Review by dsehnal — APPROVED** [review]

**Pull request merged**